### ISSUE TITLE
Trigger event on load

### ISSUE THREAD

**Issue by GeoSot** [jquery/jquery#4749]
Hello 

I am witting for first time here.
I use this loved library more than 7 years and always is a handful tool

Almost on every usage I face one MAJOR problem.
The disgusting : `Uncaught ReferenceError: $ is not defined`

Yes I know the $(function(){}) but works only after jQuery script tag

Many tricks have been published through years with deferred arrays but, it would be a great relief if we could use a custom simple event wherever we want in order to be safe

```js 
document.addEventListener('jQueryIsLoaded', function (e) {
    console.log('YES dude, you are safe to use the magic dollar symbol')  
   console.log($, 'JQuery is here')  
   
});
```

So, please make a think of triggering a raw event  after jq is loaded or any other clever way to ensure that anything , always, can run

**Comment by mgol** [comment]
Thanks for the report.

Events have an important drawback: they only fire once and if you attach a listener too late, the handler will never be invoked. Therefore, events are not that safer. Even worse, an event will just not fire and you won't see any error while if you just try to use `$` before jQuery is loaded, you get an immediate error you can act on.

I don't think there's a Web API that would make it more convenient.
